functions/Copy-DbaQueryStoreConfig.ps1

```diff
@@ -38,120 +54,116 @@ You should have received a copy of the GNU General Public License along with thi
 https://dbatools.io/Copy-QueryStoreConfig
 
 .EXAMPLE
-Copy-DbaQueryStoreConfig -SourceSqlServer ServerA\SQL -SourceDatabase AdventureWorks -TargetSqlServer ServerB\SQL
+Copy-DbaQueryStoreConfig -Source ServerA\SQL -SourceDatabase AdventureWorks -Destination ServerB\SQL -AllDatabases
 
 Copy the Query Store configuration of the AdventureWorks database in the ServerA\SQL Instance and apply it on all user databases in the ServerB\SQL Instance.
 
 .EXAMPLE
-Copy-DbaQueryStoreConfig -SourceSqlServer ServerA\SQL -SourceDatabase AdventureWorks -TargetSqlServer ServerB\SQL -Databases WorldWideTraders
+Copy-DbaQueryStoreConfig -Source ServerA\SQL -SourceDatabase AdventureWorks -Destination ServerB\SQL -DestinationDatabase WorldWideTraders
 
 Copy the Query Store configuration of the AdventureWorks database in the ServerA\SQL Instance and apply it to the WorldWideTraders database in the ServerB\SQL Instance.
 	
 #>
-	[CmdletBinding()]
+	[CmdletBinding(SupportsShouldProcess = $true)]
 	Param (
-		[parameter(Mandatory = $true, ValueFromPipeline = $true)][string[]]$SourceSqlServer,
-        [parameter(Mandatory = $true, ValueFromPipeline = $true)][string[]]$SourceDatabase,
-        [parameter(Mandatory = $true, ValueFromPipeline = $true)][string[]]$TargetSqlServer,
-		[PsCredential]$Credential
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$Source,
+		[System.Management.Automation.PSCredential]$SourceSqlCredential,
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object]$SourceDatabase,
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$Destination,
+		[System.Management.Automation.PSCredential]$DestinationSqlCredential,
+		[object[]]$DestinationDatabase,
+		[object[]]$Exclude,
+		[switch]$AllDatabases
 	)
-
-	DynamicParam {
-		if ($SourceSqlServer) {
-			return Get-ParamSqlDatabases -SqlServer $TargetSqlServer[0] -SqlCredential $Credential
-		}
-	}
-
+	
 	BEGIN
 	{
-		# Convert from RuntimeDefinedParameter object to regular array
-		$databases = $psboundparameters.Databases
-		$exclude = $psboundparameters.Exclude
-
-        Write-Verbose "Connecting to source: $SourceSqlServer"
-			try
-			{
-				$server = Connect-SqlServer -SqlServer $SourceSqlServer -SqlCredential $Credential
-
-			}
-			catch
-			{
-				if ($SourceSqlServer.count -eq 1)
-				{
-					throw $_
-				}
-				else
-				{
-					Write-Warning "Can't connect to $SourceSqlServer."
-					Continue
-				}
-			}
-
-        # Grab the Query Store configuration from the SourceDatabase through the Get-DbaQueryStoreConfig function
-        $SourceQSConfig = Get-DbaQueryStoreConfig -SqlServer $SourceSqlServer -Databases $SourceDatabase
-
+	
+		Write-Verbose "Connecting to source: $Source"
+		try
+		{
+			$sourceserver = Connect-SqlServer -SqlServer $Source -SqlCredential $SourceSqlCredential
```

Comment:
$Source can contain multiple objects, yet here's a connection made to $source in BEGIN{}.
Probably $Source should accept only one SqlInstance?